### PR TITLE
add APK build time benchmarks

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

A benchmark to measure the time it takes to build APKs to ensure that assemble (from the work in progress PR https://github.com/flutter/flutter/pull/32816) changes doesn't regress APK build times

### CODE REVIEW

File: dev/devicelab/lib/tasks/build_benchmarks.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      final int thirdBuildMilliseconds = stopwatch.elapsedMilliseconds;
+      stopwatch.stop();
+      final Map<String, double> allResults = <String, double>{};
+      allResults['first_build_debug_millis'] = initialBuildMilliseconds.toDouble();
+      allResults['second_build_debug_millis'] = secondBuildMilliseconds.toDouble();
+      allResults['after_config_change_build_debug_millis'] = thirdBuildMilliseconds.toDouble();
+      return TaskResult.success(allResults, benchmarkScoreKeys: allResults.keys.toList());
+    });
+  };
+}
```
Comment (goderbauer) [inline dev/devicelab/lib/tasks/build_benchmarks.dart]: nit: end with new line?

File: dev/devicelab/manifest.yaml [hunk 2]
```diff
@@ -357,6 +357,13 @@ tasks:
     required_agent_capabilities: ["linux/android"]
     flaky: true
 
+  build_benchmark:
+    description: >
+      Measures APK build performance.
```
Comment (goderbauer) [inline dev/devicelab/manifest.yaml]: Can this be slightly more specific? Isn't build performance when config changes or something like that?

**Review by goderbauer — APPROVED** [review]
LGTM

What's the assemble change? Can you link from the PR description to a PR/issue for context?

File: dev/devicelab/lib/tasks/build_benchmarks.dart (hunk 1 above)
Comment (jonahwilliams) [reply dev/devicelab/lib/tasks/build_benchmarks.dart]: Done

File: dev/devicelab/manifest.yaml (hunk 2 above)
Comment (jonahwilliams) [reply dev/devicelab/manifest.yaml]: Done

**Comment by jonahwilliams** [comment]
Updated PR with a link to the assemble changes

**Comment by tvolkert** [comment]
Please baseline these new benchmarks now that they're live.

**Comment by jonahwilliams** [comment]
Done

**Pull request merged**